app/code/Magento/CustomerGraphQl/Controller/HttpRequestProcessor/CustomerProcessor.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Controller\HttpRequestProcessor;
+
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Api\Data\GroupInterface;
+use Magento\Customer\Model\Context;
+use Magento\Customer\Model\GroupManagement;
+use Magento\GraphQl\Controller\HttpHeaderProcessorInterface;
+use Magento\GraphQl\Model\Query\ContextFactoryInterface;
+use Magento\GraphQl\Model\Query\ContextInterface as GraphQlContext;
+use Magento\Framework\App\Http\Context as HttpContext;
+
+/**
+ * Process the "Authorization" header
+ */
+class CustomerProcessor implements HttpHeaderProcessorInterface
+{
+    /**
+     * @var GraphQlContext
+     */
+    protected $graphQlContext;
```

Comment:
Please, use private scope for the properties. Introducing a possibility of extension by overriding is not recommended. You may use [recently created services](https://github.com/magento/magento2/blob/24b4dcd99161875957040ff83f53c20389461676/app/code/Magento/MediaGallery/Model/Directory/Command/DeleteByPaths.php#L19) as examples for good practices.